shell/server/proxy.js

```diff
@@ -1649,24 +1643,84 @@ Proxy.prototype.handleRequest = function (request, data, response, retryCount) {
     var path = request.url.slice(1);  // remove leading '/'
     var session = self.getSession(request);
 
-    if (request.method === "GET") {
+    function optionsForBody() {
+      return {
+        content: data,
+        encoding: request.headers["content-encoding"],
+        mimeType: request.headers["content-type"]
+      };
+    }
+
+    if (request.method === "GET" || request.method === "HEAD") {
```

Comment:
It does.

That said, we should probably be passing the head request as a head request, not as a get, because it's unnecessary work for the backend to generate and send e.g. a 2GB file if the frontend is going to discard it (and then potentially request the whole thing as a GET, now that it knows some metadata).  But that can be a separate bug.